src/perftest_resources.c

```diff
@@ -1471,43 +1312,10 @@ int destroy_ctx(struct pingpong_context *ctx,
 		}
 	}
 
-	#ifdef HAVE_CUDA
-	if (user_param->use_cuda) {
-		for (i = 0; i < dereg_counter; i++) {
-			CUdeviceptr d_A = (CUdeviceptr)ctx->buf[i];
-
-			printf("deallocating RX GPU buffer %016llx\n", d_A);
-			cuMemFree(d_A);
-			d_A = 0;
-		}
-		pp_free_gpu(ctx);
+	for (i = 0; i < dereg_counter; i++) {
+		ctx->memory->free_buffer(ctx->memory, 0, ctx->buf[i], ctx->buff_size);
```

Comment:
Are u sure thats its making the exact behaviour for mmap?, before it did munmap for only the first qp, now for all

Reply:
I think that current mmap implementation can handle only num_of_qps=1 or mr_per_qp=false cases, see line 190:
```
	ctx->buf[0] = mmap(NULL, size, PROT_WRITE | PROT_READ, MAP_SHARED, fd,
			offset);
```
So dereg_counter should be equal to 1 in that case:
```
	dereg_counter = (user_param->mr_per_qp) ? user_param->num_of_qps : 1;
```

Reply:
I think that we must make it as the current implementation and refactor that after if needed

Reply:
Checked this again and we are keeping same behavior as before for mmap.

Reply:
How? I don't see perftest limit the number of qp's in case of mmap